forms/gridfield/GridFieldExportButton.php

```diff
@@ -165,7 +164,14 @@ public function generateExportFileData($gridField) {
 			}
 		}
 
-		return $fileData;
+		// Convert the $fileData array into csv by capturing fputcsv's output
+		ob_start();
+		$fp = fopen('php://output', 'w');
+		foreach($fileData as $line) {
+			fputcsv($fp, $line, $this->csvSeparator, $this->csvEnclosure);
+		}
+		fclose($fp);
+		return ob_get_clean();
```

Comment:
`ob_start` calls can be nested though, so it should work even if people are already buffering content. But I'm happy to use that other method, it does look more elegant!